File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/services/core/entity/CoreAllElementRegisterKeyEnum.java
package org.infra.reactive.form.engine.form.engine.services.core.entity;

import java.util.Arrays;
import java.util.Optional;

public enum CoreAllElementRegisterKeyEnum {
    Window("Window"),
    Tab("Tab"),
    Field("Field"),
    Process("Process"),
    CoreProcessParam("CoreProcessParam"),
    Column("Column"),
    Button("Button"),
    Exception("Exception"),
    Dashboard("Dashboard"),
    AnalyticReport("AnalyticReport"),
    Table("Table"),
    CoreMenu("CoreMenu"),
    CoreRole("CoreRole"),
    CoreFilter("CoreFilter"),
    CoreFilterOperation("CoreFilterOperation"),
    CoreFilterOperationParam("CoreFilterOperationParam"),
    CoreFilterProvider("CoreFilterProvider"),
    CoreFilterProviderElement("CoreFilterProviderElement"),
    CoreWindowTab_Toolbar("CoreWindowTab_Toolbar"),
    CoreTableColumnDataProviderListValues("CoreTableColumnDataProviderListValues");

    private final String value;

    CoreAllElementRegisterKeyEnum(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    public static Optional<CoreAllElementRegisterKeyEnum> getEnum(String key) {
        return Arrays.stream(values()).filter(coreAllElementRegisterKeyEnum -> coreAllElementRegisterKeyEnum.value.equals(key)).findFirst();
    }
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Response/Element/CoreAllElementRegisterKeyEnum.js
export default class CoreAllElementRegisterKeyEnum {

}

CoreAllElementRegisterKeyEnum.RegisterKey = () => {
    return Object.freeze({
        Window: Symbol("Window"),
        Tab: Symbol("Tab"),
        Field: Symbol("Field"),
        CoreProcessParam: Symbol("CoreProcessParam"),
        Column: Symbol("Column"),
        Button: Symbol("Button"),
        Exception: Symbol("Exception"),
        Dashboard: Symbol("Dashboard"),
        Table: Symbol("Table"),
        AnalyticReport: Symbol("AnalyticReport"),
        ChartElement: Symbol("ChartElement"),
        InfraMainPage: Symbol("InfraMainPage"),
        CoreWindowTabButton: Symbol("CoreWindowTabButton")
    });
}